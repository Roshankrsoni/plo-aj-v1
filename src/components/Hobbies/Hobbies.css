.subsection + .subsection{
    margin-top:6rem;
}

.imgcarousel{
    display: flex;
    overflow-y: scroll;
    padding-bottom: 0.5rem;;
}
.drawings{
    width:40%;
}
.drawings + .drawings{
   margin-left: 1rem;
}



/* width */
.imgcarousel::-webkit-scrollbar {
width: 0;
height:4px;
}

/* Track */
.imgcarousel::-webkit-scrollbar-track {
background: white;
}

/* Handle */
.imgcarousel::-webkit-scrollbar-thumb {
background: #f2f2f2;
border-radius: 10px;
}

/* Handle on hover */
.imgcarousel::-webkit-scrollbar-thumb:hover {
background: #333333;
}


/* Track */
.darkmode .imgcarousel::-webkit-scrollbar-track {
background: #333333;

}

/* Handle */
.darkmode  .imgcarousel::-webkit-scrollbar-thumb {
background: #4F4F4F;
border-radius: 10px;
}

/* Handle on hover */
.darkmode  .imgcarousel::-webkit-scrollbar-thumb:hover {
background: #333333;
}
