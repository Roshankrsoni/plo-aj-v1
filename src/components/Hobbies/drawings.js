function importAll(r) {
    return r.keys().map(r);
}

const drawings = importAll(require.context('../../assets/drawings/', false, /\.(png|jpe?g|svg)$/));

export default drawings;