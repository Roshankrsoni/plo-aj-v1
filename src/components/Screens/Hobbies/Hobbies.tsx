import React from "react";
import { useGlobalContext } from "../../../context/GlobalContext";
import constants from "../../../helpers/constants";
import T from "../../../translations/en_IN";
import SectionWrapper from "../../Organisms/SectionWrapper/SectionWrapper";
function Hobbies() {
  const { profileData } = useGlobalContext();
  const { hobbies } = profileData;
  return (
    <>
      <div className="section-title h2 bold">{T.HOBBIES_AND_INTERESTS}</div>
      <div className="subsection">
        {hobbies.map((hobby, key) => {
          return (
            <div className="subsection-data" key={String(key)}>
              <span className="h3 grey1 bold">{hobby.name}</span>
              <div className="mt-2 grey-1 body-text">{hobby.summary}</div>
            </div>
          );
        })}
      </div>
    </>
  );
}

export default SectionWrapper(Hobbies, constants.classNames.HOBBIES);
