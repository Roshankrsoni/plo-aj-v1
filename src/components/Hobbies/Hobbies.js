import React from "react";
import "./Hobbies.css";
function Hobbies(props) {
  return (
    <div
      className={
        "show-on-scroll col-md-7 page-1 text-left section hobbies-interests " +
        props.class
      }
    >
      <div className="section-title grey4 h2 bold">Hobbies & Interests</div>
      <div className="subsection">
        {props.dbData.hobbies.map((ele, key) => {
          return (
            <div className="subsection-data" key={key}>
              <span className="h3 grey1 bold">{ele.name}</span>
              <div className="mt-2 grey-1 body-text">{ele.summary}</div>
            </div>
          );
        })}
      </div>
    </div>
  );
}

export default Hobbies;
