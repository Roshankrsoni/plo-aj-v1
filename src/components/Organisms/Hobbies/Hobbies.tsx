import React from "react";
import "./Hobbies.css";
import { useGlobalContext } from "../../../context/GlobalContext";
import SectionVisibilityHOC from "../SectionVisibilityHOC/SectionVisibilityHOC";
function Hobbies() {
  const { profileData } = useGlobalContext();
  const { hobbies } = profileData;
  return (
    <>
      <div className="section-title h2 bold">Hobbies & Interests</div>
      <div className="subsection">
        {hobbies.map((hobby, key) => {
          return (
            <div className="subsection-data" key={key}>
              <span className="h3 grey1 bold">{hobby.name}</span>
              <div className="mt-2 grey-1 body-text">{hobby.summary}</div>
            </div>
          );
        })}
      </div>
    </>
  );
}

export default SectionVisibilityHOC(Hobbies, "hobbies-interests");
