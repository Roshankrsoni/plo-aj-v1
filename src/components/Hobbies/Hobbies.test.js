import React from 'react';
import { shallow } from 'enzyme';
import Hobbies from './Hobbies';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Hobbies dbData={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});