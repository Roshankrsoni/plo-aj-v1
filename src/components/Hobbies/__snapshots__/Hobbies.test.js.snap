// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="show-on-scroll col-md-7 page-1 text-left section hobbies-interests undefined"
>
  <div
    className="section-title grey4 h2 bold"
  >
    Hobbies & Interests
  </div>
  <div
    className="subsection"
  >
    <div
      className="subsection-data"
      key="0"
    >
      <span
        className="h3 grey1 bold"
      >
        Drawing
      </span>
      <div
        className="mt-2 grey-1 body-text"
      >
        Since my childhood I was always passionate about drawing. I really enjoyed Art & Crafts classes in my school. I loved drawing Pokemon characters, cars and bikes. Later I was lost in the world of competition but still managed to keep my interest as we had engineering drawing, AutoCAD in the curriculum. Now I have started learning digital drawing. Please check my instagram profile for some of my work.
      </div>
    </div>
    <div
      className="subsection-data"
      key="1"
    >
      <span
        className="h3 grey1 bold"
      >
        Trekking
      </span>
      <div
        className="mt-2 grey-1 body-text"
      >
        Usually in the monsoons, I prefer going for small treks to take a break from work and the daily routine. The colossal mountains and lush green pastures always fascinates me. Being surrounded by mountains and trees always revives and rejuvenates me. I have completed various treks in the Western ghats/Sahyadri region like Torna, Rajmachi, Sudhagad, Korigad, Kalsubai, Vikat Gad, Mahuli Gad, Visapur, Sihagad, Vasota jungle trek, Tung & Tikona, Tadiandamol trek and the mighty Katraj to Sihagad night trek.
      </div>
    </div>
    <div
      className="subsection-data"
      key="2"
    >
      <span
        className="h3 grey1 bold"
      >
        Bike Riding
      </span>
      <div
        className="mt-2 grey-1 body-text"
      >
        From the day I have bought the RE classic 350 I have developed this unstoppable urge of going new places riding my bike. I usually travel Mumbai - Pune on bike. The Most memorable bike trip would be Leh Ladakh - Khardungla and Changla Pass. The stunning view, endless empty roads full of twist and turns. Also the steep ascends, and bad road condition makes Chang La and Khardungla one of the most challenging roads to drive upon.
      </div>
    </div>
  </div>
</div>
`;
